rb/router.py

```diff
@@ -132,8 +133,8 @@ def __init__(self, cluster):
         assert_gapless_hosts(self.cluster.hosts)
 
     def get_host_for_key(self, key):
-        if isinstance(key, unicode):
+        if isinstance(key, text_type):
             k = key.encode('utf-8')
         else:
-            k = str(key)
+            k = bytes_type(key)
```

Comment:
Will `key` always be a string or bytes? If it's an integer then using `bytes` here in python 3 will be bad. I feel like the intent of the `str` here was "Make sure this is a string", in terms of converting ints to a string (which effectively means "ascii encoded bytes", I suppose).

Reply:
I've guarded against integer types and coerced them into strings now. I left the byte conversion fallback.